test/integration/h2_fuzz.cc

```diff
@@ -0,0 +1,249 @@
+    const Http2Frame::ErrorCode error_code =
+        static_cast<Http2Frame::ErrorCode>(proto_frame.reset_stream().error_code());
+    ENVOY_LOG_MISC(trace, "Sending reset stream frame");
+    h2_frame = Http2Frame::makeResetStreamFrame(stream_idx, error_code);
+    break;
+  }
+  case test::integration::H2TestFrame::kGoAway: {
+    const uint32_t last_stream_idx = proto_frame.go_away().last_stream_index();
+    const Http2Frame::ErrorCode error_code =
+        static_cast<Http2Frame::ErrorCode>(proto_frame.go_away().error_code());
+    ENVOY_LOG_MISC(trace, "Sending go-away frame");
+    h2_frame = Http2Frame::makeEmptyGoAwayFrame(last_stream_idx, error_code);
+    break;
+  }
+  case test::integration::H2TestFrame::kWindowUpdate: {
+    const uint32_t stream_idx = proto_frame.window_update().stream_index();
+    const uint32_t increment = proto_frame.window_update().increment();
+    ENVOY_LOG_MISC(trace, "Sending windows_update frame");
+    h2_frame = Http2Frame::makeWindowUpdateFrame(stream_idx, increment);
+    break;
+  }
+  case test::integration::H2TestFrame::kMalformedRequest: {
+    const uint32_t stream_idx = proto_frame.malformed_request().stream_index();
+    ENVOY_LOG_MISC(trace, "Sending malformed_request frame");
+    h2_frame = Http2Frame::makeMalformedRequest(stream_idx);
+    break;
+  }
+  case test::integration::H2TestFrame::kMalformedRequestWithZerolenHeader: {
+    const uint32_t stream_idx = proto_frame.malformed_request_with_zerolen_header().stream_index();
+    const absl::string_view host = proto_frame.malformed_request_with_zerolen_header().host();
+    const absl::string_view path = proto_frame.malformed_request_with_zerolen_header().path();
+    ENVOY_LOG_MISC(trace, "Sending malformed_request_with_zerolen_header");
+    h2_frame = Http2Frame::makeMalformedRequestWithZerolenHeader(stream_idx, host, path);
+    break;
+  }
+  case test::integration::H2TestFrame::kRequest: {
+    const uint32_t stream_idx = proto_frame.request().stream_index();
+    const absl::string_view host = proto_frame.request().host();
+    const absl::string_view path = proto_frame.request().path();
+    ENVOY_LOG_MISC(trace, "Sending request");
+    h2_frame = Http2Frame::makeRequest(stream_idx, host, path);
+    break;
+  }
+  case test::integration::H2TestFrame::kPostRequest: {
+    const uint32_t stream_idx = proto_frame.post_request().stream_index();
+    const absl::string_view host = proto_frame.post_request().host();
+    const absl::string_view path = proto_frame.post_request().path();
+    ENVOY_LOG_MISC(trace, "Sending post request");
+    h2_frame = Http2Frame::makePostRequest(stream_idx, host, path);
+    break;
+  }
+  case test::integration::H2TestFrame::kGeneric: {
+    const absl::string_view frame_bytes = proto_frame.generic().frame_bytes();
+    ENVOY_LOG_MISC(trace, "Sending generic frame");
+    h2_frame = Http2Frame::makeMalformedFrame(frame_bytes);
```

Comment:
Not sure the connection between generic and malformed is clear here.

Reply:
Renamed to makeGenericFrame, and added a method description.